openmdao/recorders/recorders.py

```diff
@@ -0,0 +1,28 @@
+from openmdao.core.options import OptionsDictionary
```

Comment:
This whole bit seems needlessly complicated and obfuscated. It could just all be in the BaseRecorder.  The only recorder that doesn't support MetaDataRecorder is the DumpRecorder, and there isn't any reason why it couldn't support that. This means that all these options and both methods can be in the base class.